### PR TITLE
Update dbt documentation to document Python models

### LINKED ISSUE #497: Document Python models
We've done enough work on Python models now that we're feeling confident we'll keep using them. Let's add some brief docs to the dbt README explaining how, when, and why we use Python models, with an eye toward helping other folks write new Python models. We should also take a second look at the issue template to see if it needs anything related to Python models.

### PR DESCRIPTION
**Pull request by jeancochrane**
This PR updates our dbt documentation in `dbt/README.md` and the `new-dbt-model` issue template to add guidance on creating Python models.

Closes #497.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/new-dbt-model.md [hunk 1]
```diff
@@ -35,14 +38,18 @@ Otherwise, delete it in favor of the long checklist in the following section.)_
 - [ ] Confirm that a subdirectory for this model's database exists in
   the `dbt/models/` directory, and if not, create one, add a new `schema.yml`
   file, and update `dbt_project.yml` to document the `+schema`
-- [ ] Define the SQL query that creates the model in the model subdirectory,
-  following any existing file naming schema
-- [ ] Optionally configure model materialization within the query file
+- [ ] Define the SQL query or Python script that creates the model in the model
+  subdirectory, following any existing file naming schema
+- [ ] Use `source()` and `ref()` to reference other models where possible
```
Comment (jeancochrane) [inline .github/ISSUE_TEMPLATE/new-dbt-model.md:43]: This line was present in the expanded checklist but missing from the condensed checklist, so I added it in.

File: .github/ISSUE_TEMPLATE/new-dbt-model.md [hunk 2]
```diff
@@ -109,35 +119,63 @@ join {{ ref('default.vw_pin_universe') }}
...
+--- dbt/models/default/default.new_model.sql
++++ dbt/models/default/default.new_model.sql
++ {{
++   config(
++     materialized='table',
++     partitioned_by=['year'],
++     bucketed_by=['pin10'],
++     bucket_count=1
++   )
++ }}
```
Comment (jeancochrane) [inline .github/ISSUE_TEMPLATE/new-dbt-model.md:152]: Changing this to a diff to make it clearer to the reader which portion of the code has been added as part of this step.

File: .github/ISSUE_TEMPLATE/new-dbt-model.md [hunk 3]
```diff
@@ -157,6 +195,53 @@ using (pin10, year)
      data_tests:
 ```
 
+- [ ] _[Python models only]_ If you need any third-party pure Python packages
```
Comment (dfsnow) [inline .github/ISSUE_TEMPLATE/new-dbt-model.md:198]: **suggestion (non-blocking):** I would maybe just add a note here linking README section that details the format used to specify dependencies.

**Review by dfsnow — APPROVED** [review]
This looks great @jeancochrane! Super detailed and helpful

File: .github/ISSUE_TEMPLATE/new-dbt-model.md [hunk 4]
```diff
@@ -109,35 +119,63 @@ join {{ ref('default.vw_pin_universe') }}
...
 ```
 
+```diff
+# Python model example
+--- dbt/models/default/default.new_model.py
++++ dbt/models/default/default.new_model.py
+import pandas as pd
+
+def model(dbt, spark_session):
++    dbt.config(materialized="table")
```
[the remark this answers is not in the document]
Comment (jeancochrane) [reply .github/ISSUE_TEMPLATE/new-dbt-model.md:164]: Good question! Looks like [dbt says that](https://docs.getdbt.com/docs/build/python-models#materializations) "Python models can't be materialized as view or ephemeral", but all of their examples in the docs include this line. My guess is that this materialization line is still required, but I'll run a test to confirm that.

File: .github/ISSUE_TEMPLATE/new-dbt-model.md (hunk 3 above)
Comment (jeancochrane) [reply .github/ISSUE_TEMPLATE/new-dbt-model.md:196]: Good call, done in [a8ed694](https://github.com/ccao-data/data-architecture/commit/a8ed694464bf18cb0df1b412b08f696b3fb5a3f5).

File: .github/ISSUE_TEMPLATE/new-dbt-model.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (jeancochrane) [reply .github/ISSUE_TEMPLATE/new-dbt-model.md:164]: Looks like the model still works if you remove the `dbt.config(materialized="table")` line, so I'll go ahead and remove that from the docs and from the `reporting.ratio_stats` model!

[the remark this answers is not in the document]
Comment (jeancochrane) [reply .github/ISSUE_TEMPLATE/new-dbt-model.md]: Done in 22226d27e2e7e11e45995a0631f4d927d34b31d1 and eaf88044e317f804f5f39e7da5b2c08f6673d4a0.

[the remark this answers is not in the document]
Comment (jeancochrane) [reply .github/ISSUE_TEMPLATE/new-dbt-model.md]: Actually, I undid the change to `ratio_stats` in 009c2239d8860318e7c8325a33e1df50dcf5192f. We should do it eventually but I don't want to have to bother with rebuilding the table right now.

**Pull request merged**